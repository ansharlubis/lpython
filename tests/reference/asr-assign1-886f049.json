{
    "basename": "asr-assign1-886f049",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/assign1.py",
    "infile_hash": "3b82a73e457bd65e85828b72d56596ca927e7c661e333691f154912b",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-assign1-886f049.stdout",
    "stdout_hash": "b2525f03f0a222d996b40151a539a1c20bf18dbc0207820817139951",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}