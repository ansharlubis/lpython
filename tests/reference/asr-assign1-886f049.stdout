([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         test_augassign: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           a [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mCharacter[39m[0m 1 -2 () []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     r: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           r [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     s: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           s [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               test_augassign [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mr[39m) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mr[39m) 
                  ([1m[35mIntegerBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mr[39m) [1m[32m
                     Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m4[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mInteger[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mr[39m) 
                  ([1m[35mIntegerBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mr[39m) [1m[32m
                     Mul[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) 
                     ([1m[35mInteger[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mr[39m) 
                  ([1m[35mIntegerBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mr[39m) [1m[32m
                     Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mInteger[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) ([1m[35mIntegerConstant[39m[0m [36m10[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mr[39m) 
                  ([1m[35mRealBinOp[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mr[39m) [1m[32m
                        IntegerToReal[39m[0m 
                        ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                     Div[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33ms[39m) [1m[32m
                        IntegerToReal[39m[0m 
                        ([1m[35mReal[39m[0m 8 []) ()) 
                     ([1m[35mReal[39m[0m 8 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mStringConstant[39m[0m "" 
                     ([1m[35mCharacter[39m[0m 1 0 () [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mStringConcat[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ma[39m) 
                     ([1m[35mStringConstant[39m[0m "test" 
                        ([1m[35mCharacter[39m[0m 1 4 () [])) 
                     ([1m[35mCharacter[39m[0m 1 2 () []) ()) ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [])
(TranslationUnit (SymbolTable 1 {main_program: (Program (SymbolTable 3 {}) main_program [] []), test_augassign: (Subroutine (SymbolTable 2 {a: (Variable 2 a Local () () Default (Character 1 -2 () []) Source Public Required .false.), r: (Variable 2 r Local () () Default (Integer 4 []) Source Public Required .false.), s: (Variable 2 s Local () () Default (Integer 4 []) Source Public Required .false.)}) test_augassign [] [(= (Var 2 r) (IntegerConstant 0 (Integer 4 [])) ()) (= (Var 2 r) (IntegerBinOp (Var 2 r) Add (IntegerConstant 4 (Integer 4 [])) (Integer 4 []) ()) ()) (= (Var 2 s) (IntegerConstant 5 (Integer 4 [])) ()) (= (Var 2 r) (IntegerBinOp (Var 2 r) Mul (Var 2 s) (Integer 4 []) ()) ()) (= (Var 2 r) (IntegerBinOp (Var 2 r) Sub (IntegerConstant 2 (Integer 4 [])) (Integer 4 []) ()) ()) (= (Var 2 s) (IntegerConstant 10 (Integer 4 [])) ()) (= (Var 2 r) (RealBinOp (Cast (Var 2 r) IntegerToReal (Real 8 []) ()) Div (Cast (Var 2 s) IntegerToReal (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 a) (StringConstant "" (Character 1 0 () [])) ()) (= (Var 2 a) (StringConcat (Var 2 a) (StringConstant "test" (Character 1 4 () [])) (Character 1 2 () []) ()) ())] Source Public Implementation () .false. .false.)}) [])
